crates/m3/src/builder/table.rs

```diff
@@ -258,6 +258,81 @@ impl<'a, F: TowerField> TableBuilder<'a, F> {
 		)
 	}
 
+	/// Adds field exponentiation column with a fixed base
+	///
+	/// ## Parameters
+	/// - `name`: Name for the column
+	/// - `pow_bits`: The bits of exponent columns from LSB to MSB
+	/// - `base`: The base to exponentiate. The field used in exponentiation will be `FSub`
+	///
+	/// ## Preconditions
+	/// * `pow_bits.len()` must be a power of 2 and less than or equal to the width of field `FSub`
+	///
+	/// ## NOTE
+	/// * The witness generation for the return column will be done inside gkr_gpa *
+	pub fn add_static_exp<FSub>(
+		&mut self,
+		name: impl ToString,
+		pow_bits: &[Col<B1>],
+		base: F,
+	) -> Col<FSub>
+	where
+		FSub: TowerField,
+		F: ExtensionField<FSub>,
+	{
+		assert!(pow_bits.len().is_power_of_two());
+		assert!(pow_bits.len() <= 1 << (FSub::TOWER_LEVEL));
+
+		// TODO: Add check for pow_bits, F, FSub, VALUES_PER_ROW
+		let namespaced_name = self.namespaced_name(name);
+		let bit_cols = pow_bits.iter().map(|bit| bit.id().table_index).collect();
+		self.table.new_column(
+			namespaced_name,
+			ColumnDef::StaticExp {
+				bit_cols,
+				base,
+				base_tower_level: FSub::TOWER_LEVEL,
+			},
+		)
+	}
+
+	/// Adds field exponentiation column with a base from another column
+	///
+	/// ## Parameters
+	/// - `name`: Name for the column
+	/// - `pow_bits`: The bits of exponent columns from LSB to MSB
+	/// - `base`: The column of base to exponentiate. The field used in exponentiation will be `FSub`
+	///
+	/// ## Preconditions
+	/// * `pow_bits.len()` must be a power of 2 and less than or equal to the width of field `FSub`
+	///
+	/// ## NOTE
+	/// * The witness generation for the return column will be done inside gkr_gpa *
+	pub fn add_dynamic_exp<FSub>(
+		&mut self,
+		name: impl ToString,
+		pow_bits: &[Col<B1>],
+		base: Col<FSub>,
+	) -> Col<FSub>
+	where
+		FSub: TowerField,
+		F: ExtensionField<FSub>,
+	{
+		assert!(pow_bits.len().is_power_of_two());
+		assert!(pow_bits.len() <= (1 << FSub::TOWER_LEVEL));
+
+		let namespaced_name = self.namespaced_name(name);
+		let bit_cols = pow_bits.iter().map(|bit| bit.id().table_index).collect();
+		self.table.new_column(
+			namespaced_name,
+			ColumnDef::DynamicExp {
+				bit_cols,
+				base: base.id().table_index,
+				base_tower_level: FSub::TOWER_LEVEL,
```

Comment:
`FSub::TOWER_LEVEL` is very expensive, you should add FExpBase (see old circuit)